quinn-h3/src/client.rs

```diff
@@ -137,12 +272,67 @@ impl Client {
     }
 }
 
+/// HTTP/3 handshake future
+///
+/// Represents an ongoing HTTP/3 handshake. Upon success, this future will resolve into an
+/// eshtablished [`Connection`].
+///
+/// [`Connection`]: struct.Connection.html
 pub struct Connecting {
     connecting: quinn::Connecting,
     settings: Settings,
 }
 
 impl Connecting {
+    /// Try to convert an ongoing handshake into a 0-RTT enabled exchange.
+    ///
+    /// # About 0-RTT
+    ///
+    /// 0 Round Trip Time is a QUIC feature enabling application data exchange before the
+    /// peers have finished the TLS handshake. It's based on prior crypto exchange, reused
+    /// for a new connection. So this function will fail if it's the first contact with
+    /// the server.
+    ///
+    /// # Security concerns
+    ///
+    /// This functionality is vulnerable to replay attacks. Therefore any request with a
+    /// non-idempotent method sent before [`ZeroRttAccepted`] resolves will fail.
+    ///
+    /// # Usage
+    ///
+    /// Upon success, this method returns a [`Connection`], and a `Future`
+    /// that will resolve when the handshake completes: [`ZeroRttAccepted`]. If this peer
+    /// is not associated with any known prior connection, the [`Connecting`] handshake
+    /// completion future will be handed back.
+    ///
+    /// ## Task spawing
+    ///
+    /// This method spawns two driver tasks for connection management at QUIC and HTTP/3
```

Comment:
About the implementation: couldn't we spawn a single task here that drives both?

Reply:
So they would be driven sequentially? Or be nested?

This would bring some more control on the sequence of events happening to a connection.

Reply:
Not sure what the best way forward would be. But using two tasks seems potentially wasteful. @Ralith any thoughts?